### PR TITLE
Issue #6811 - Clarify event selectability in competition registration form

### PR DESCRIPTION
**Pull request by revers3ntropy**
Clarify event selectability in competition registration form.

Adds a hint above the cube icons when no events are selected at first only on registration forms.

### CODE REVIEW

File: WcaOnRails/app/views/shared/_associated_events_picker.html.erb [hunk 1]
```diff
@@ -4,15 +4,17 @@ Variables:
  - disabled
  - events_association_name -> name of the associated events relation
  - allowed_events -> a relation or an Array of Event objects that should be shown (all by default)
- - selected_events -> a relation or an Array of Event objects that whould be marked as selected
+ - selected_events -> a relation or an Array of Event objects that would be marked as selected
```
Comment (gregorbg) [inline WcaOnRails/app/views/shared/_associated_events_picker.html.erb:7]: Nice catch! 😄

**Review by gregorbg — COMMENTED** [review]
Thanks for your contribution! One remark about the logic of displaying the box, other than that all good 👍

File: WcaOnRails/app/views/shared/_associated_events_picker.html.erb [hunk 2]
```diff
@@ -31,6 +33,10 @@ Note: form_builder.object is the object having associated events.
     <% end %>
   <% end %>
   <div id="<%= events_association_name %>" class="associated-events">
+    <% if show_select_hint && (selected_events.length == 0) %>
```
[the remark this answers is not in the document]
Comment (revers3ntropy) [reply WcaOnRails/app/views/shared/_associated_events_picker.html.erb:36]: Ok I've implemented this now

File: WcaOnRails/app/assets/javascripts/registrations.js [hunk 3]
```diff
@@ -38,24 +38,24 @@ onPage('registrations#index', function() {
...
-    return paidA === "" ? 1 : -1;
   }
+  return paidA === "" ? 1 : -1;
 }
 
 function compareHtmlContent(a, b) {
-  var first = $('<p>' + a + '</p>').text().trim();
-  var second = $('<p>' + b + '</p>').text().trim();
+  const first = $('<p>' + a + '</p>').text().trim();
+  const second = $('<p>' + b + '</p>').text().trim();
```
Comment (gregorbg) [inline WcaOnRails/app/assets/javascripts/registrations.js:58]: Seems like your linter tapped into unrelated code?

File: WcaOnRails/app/assets/javascripts/registrations.js [hunk 4]
```diff
@@ -68,3 +68,12 @@ onPage('registrations#add, registrations#do_add', function() {
     $('#events input[type="checkbox"]').prop('checked', true);
   });
 });
+
+onPage('registrations#create', function() {
+  // Hide the hint when the user selects an event
+  // or selects all events
+  $('.associated-events input[type="checkbox"], .select-all-events').click(function() {
+    // opacity:0 rather than display:none to avoid DOM shifting
+    $('.associated-events .select-hint').css('opacity', '0');
```
Comment (gregorbg) [inline WcaOnRails/app/assets/javascripts/registrations.js:77]: What about `visibility:hidden`?

File: WcaOnRails/app/assets/javascripts/registrations.js [hunk 5]
```diff
@@ -24,7 +24,7 @@ onPage('registrations#edit_registrations', function() {
   showHideActions();
 
   $('button[value=delete-selected]').on("click", function(e) {
-    var $selectedRows = $registrationsTable.find("tr.selected");
+    const $selectedRows = $registrationsTable.find("tr.selected");
```
Comment (gregorbg) [inline WcaOnRails/app/assets/javascripts/registrations.js:27]: Linter touching unrelated code, see my other block comment below.

**Review by gregorbg — CHANGES_REQUESTED** [review]

File: WcaOnRails/app/assets/javascripts/registrations.js (hunk 3 above)
Comment (revers3ntropy) [reply WcaOnRails/app/assets/javascripts/registrations.js]: Whoops! I meant to roll those changes back before commiting sorry! Think I've fixed it now...

File: WcaOnRails/app/assets/javascripts/registrations.js (hunk 4 above)
Comment (revers3ntropy) [reply WcaOnRails/app/assets/javascripts/registrations.js]: Ah is that prefered? I default to use opacity but I guess visibility is more readable.
